client/src/app/services/hooks/use-manage-play.ts

```diff
@@ -1,188 +1,203 @@
-        const response = await getLastPlayByAccountHash(accountHash);
-
-        const play = response.data[0] as any;
-
-        if (play.deployHash === deploy.deploy_hash) {
-          setPlayResult({
-            data: play,
-            loading: false,
-            error: false,
-          });
-          reloadPlaysData();
-        } else {
-          throw new Error('A new play was not created');
-        }
-      } catch (error) {
-        setPlayResult(errorPlayResult);
-      }
-    } else {
-      console.error(`Deploy failed: ${deploy.error_message}`);
-      setPlayResult({
-        ...playResult,
-        data: DeployFailed.Failed,
-        loading: false,
-      });
-    }
-  };
-
-  const endPlaying = () => {
-    if (!executedDeploy) {
-      setPlayResult(errorPlayResult);
-    } else {
-      setExecutedDeploy(null);
-    }
-
-    closePlayerDeploysWebSocketStream();
-  }
-
-  const connectWallet = async () => {
-    await clickRef?.signIn();
-  };
-
-  return {
-    playerAccount,
-    connectWallet,
-    startPlaying,
-    endPlaying,
-    playResult
-  };
+	const clickRef = useClickRef();
+	const activeAccountContext = useContext(ActiveAccountContext);
+
+	const [playerAccount, setPlayerAccount] =
+		useState<AccountType | null>(null);
+
+	const [playResult, setPlayResult] = useState<PlayResult>({
+		data: null,
+		loading: false,
+		cancelled: false,
+		error: false,
+	});
+
+	const errorPlayResult = () => ({
+		data: null,
+		loading: false,
+		cancelled: false,
+		error: true,
+	});
+
+	const cancelledPlayResult = () => ({
+		data: null,
+		loading: false,
+		cancelled: true,
+		error: false,
+	});
+
+	const [executedDeploy, setExecutedDeploy] = useState<Deploy | null>(
+		null
+	);
+
+	const { reloadPlaysData } = usePlaysData();
+
+	useEffect(() => {
+		if (activeAccountContext && clickRef) {
+			clickRef
+				.getActiveAccountAsync({ withBalance: true })
+				.then(playerAccount => {
+					setPlayerAccount(playerAccount);
+				});
+		} else {
+			setPlayerAccount(null);
+		}
+	}, [activeAccountContext]);
+
+	useEffect(() => {
+		if (executedDeploy !== null) {
+			handleDeployProcessed(executedDeploy);
+		}
+	}, [executedDeploy]);
+
+	const handleDeployStatusUpdate = async (
+		status: string,
+		data: any
+	) => {
+		if (status === TransactionStatus.SENT) {
+			setPlayResult({
+				data: null,
+				loading: true,
+				cancelled: false,
+				error: false,
+			});
+			setExecutedDeploy(null);
+		} else if (status === TransactionStatus.CANCELLED) {
+			setPlayResult(cancelledPlayResult());
+		} else if (
+			status === TransactionStatus.TIMEOUT ||
+			status === TransactionStatus.ERROR
+		) {
+			setPlayResult(errorPlayResult());
+		} else if (status === TransactionStatus.PROCESSED) {
+			setExecutedDeploy(data.csprCloudTransaction);
+		}
+	};
+
+	const startPlaying = async () => {
+		if (!playerAccount?.public_key) {
+			setPlayResult(errorPlayResult());
+			return;
+		}
+
+		const playerPublicKey = CLPublicKey.fromHex(
```

Comment:
This seems to be the old SDK. If the exposure is minimal can change to the latest one in this PR or in a separate one?